alphagov/notifications-admin
# Service join request approve and set permissions

For service-join-request when approver clicks on the link from email, it takes user on Yes/No options. This PR is for when user selects Yes then user is asked to set permissions for requester. If user selects No then is shows refused message. More details on [this card](https://trello.com/c/IdQPhhuy/1000-update-the-servicejoinrequest-when-request-is-approved-rejected).

Approver will be asked to set permissions for the requester:
![image](https://github.com/user-attachments/assets/6cb1a5fe-ea1e-428d-97a9-001fc76397c0)

https://github.com/user-attachments/assets/044b8f38-8cd2-475e-bef0-26d080c11679



Approver will be asked to inform the requester on refuse:
![image](https://github.com/user-attachments/assets/58e12e5e-fc2e-4aab-ad82-3b37019cc675)

https://github.com/user-attachments/assets/c2620194-ee5d-4304-aabf-ead60e4af116




**Steps to test locally:**
This card is only for notify-admin changes. But you will require specific notify-api changes.
Check out this branch
Check out notify-api : https://github.com/alphagov/notifications-api/tree/feat/update-service-join-request (optional: and merge main into this branch)

*Test organisation and test users*
You will need Organisation with “Users can ask to join service” enabled
You will need two users:
Approver: who has manage-users access for any live service
Requester: who is in the org without live service access

*From Requester account:*
- Request to join a live service
- In `service_join_requests` table one record will be added or you can add manually if you want skip above steps
- Approver URL: http://notify.localhost:6012/services/<service_join_requests.service_id>/join-request/<service_join_requests.id>/approve
- Update ids in above URL
- Update the request status from approved/rejected to pending manually in db to skip above steps for other scenarios
